pr_agent/algo/pr_processing.py

```diff
@@ -43,9 +44,8 @@ def get_pr_diff(git_provider: GitProvider, token_handler: TokenHandler, model: s
     """
 
     if disable_extra_lines:
+        global PATCH_EXTRA_LINES
         PATCH_EXTRA_LINES = 0
```

Comment:
**Security Issue**: The use of a global variable to control the behavior of a function can lead to unpredictable outcomes and makes the code harder to understand and maintain. Moreover, modifying global state can introduce side effects that are difficult to track and debug. <br> **Fix**: Refactor the function to accept PATCH_EXTRA_LINES as a parameter instead of relying on a global variable. This approach improves the function's predictability and reusability. <br> **Code Suggestion**: 
 ```
 def get_pr_diff(git_provider: GitProvider, token_handler: TokenHandler, model: str,
                  add_line_numbers_to_hunks: bool = False, disable_extra_lines: bool = False, PATCH_EXTRA_LINES: int = 3) -> str:
     if disable_extra_lines:
         PATCH_EXTRA_LINES = 0
     else:
         PATCH_EXTRA_LINES = get_settings().config.patch_extra_lines

     try:
         diff_files = git_provider.get_diff_files()
         ...
     except Exception as e:
         logging.error("Failed to generate PR diff: ", e)
         raise
 ```